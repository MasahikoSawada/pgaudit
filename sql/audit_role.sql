CREATE USER audit_role SUPERUSER;
CREATE USER unaudit_role;

-- Operations by unaudit_role should not be logged.
\c bar unaudit_role
CREATE TABLE role_test(c int primary key);
INSERT INTO role_test VALUES(1);
SELECT * FROM role_test;
DROP TABLE role_test;

-- Operation by audit_role should be logged
-- according to configuration. DDL and READ
-- are logged.
\c bar audit_role
-- logged
CREATE TABLE role_test(c int primary key);
-- not logged
INSERT INTO role_test VALUES(1);
-- logged
SELECT * FROM role_test;
-- logged
DROP TABLE role_test;

-- SET ROLE doesn't work to avoid emit
SET ROLE unaudit_role;
CREATE TABLE role_test(c int primary key);
INSERT INTO role_test VALUES(1);
SELECT * FROM role_test;
DROP TABLE role_test;
