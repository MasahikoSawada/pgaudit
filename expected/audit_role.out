CREATE USER audit_role SUPERUSER;
CREATE USER unaudit_role;
-- Operations by unaudit_role should not be logged.
\c bar unaudit_role
CREATE TABLE role_test(c int primary key);
INSERT INTO role_test VALUES(1);
SELECT * FROM role_test;
 c 
---
 1
(1 row)

DROP TABLE role_test;
-- Operation by audit_role should be logged
-- according to configuration. DDL and READ
-- are logged.
\c bar audit_role
-- logged
CREATE TABLE role_test(c int primary key);
NOTICE:  AUDIT: SESSION,1,1,DDL,,bar,[local],pg_regress,,,,,CREATE TABLE,TABLE,public.role_test,CREATE TABLE role_test(c int primary key);,<not logged>
NOTICE:  AUDIT: SESSION,1,1,DDL,,bar,[local],pg_regress,,,,,CREATE INDEX,INDEX,public.role_test_pkey,CREATE TABLE role_test(c int primary key);,<not logged>
-- not logged
INSERT INTO role_test VALUES(1);
-- logged
SELECT * FROM role_test;
NOTICE:  AUDIT: SESSION,2,1,READ,,bar,[local],pg_regress,,,,,SELECT,TABLE,public.role_test,SELECT * FROM role_test;,<not logged>
 c 
---
 1
(1 row)

-- logged
DROP TABLE role_test;
NOTICE:  AUDIT: SESSION,3,1,DDL,,bar,[local],pg_regress,,,,,DROP TABLE,TABLE,public.role_test,DROP TABLE role_test;,<not logged>
NOTICE:  AUDIT: SESSION,3,1,DDL,,bar,[local],pg_regress,,,,,DROP TABLE,TABLE CONSTRAINT,role_test_pkey on public.role_test,DROP TABLE role_test;,<not logged>
NOTICE:  AUDIT: SESSION,3,1,DDL,,bar,[local],pg_regress,,,,,DROP TABLE,INDEX,public.role_test_pkey,DROP TABLE role_test;,<not logged>
-- SET ROLE doesn't work to avoid emit
SET ROLE unaudit_role;
CREATE TABLE role_test(c int primary key);
NOTICE:  AUDIT: SESSION,4,1,DDL,,bar,[local],pg_regress,,,,,CREATE TABLE,TABLE,public.role_test,CREATE TABLE role_test(c int primary key);,<not logged>
NOTICE:  AUDIT: SESSION,4,1,DDL,,bar,[local],pg_regress,,,,,CREATE INDEX,INDEX,public.role_test_pkey,CREATE TABLE role_test(c int primary key);,<not logged>
INSERT INTO role_test VALUES(1);
SELECT * FROM role_test;
NOTICE:  AUDIT: SESSION,5,1,READ,,bar,[local],pg_regress,,,,,SELECT,TABLE,public.role_test,SELECT * FROM role_test;,<not logged>
 c 
---
 1
(1 row)

DROP TABLE role_test;
NOTICE:  AUDIT: SESSION,6,1,DDL,,bar,[local],pg_regress,,,,,DROP TABLE,TABLE,public.role_test,DROP TABLE role_test;,<not logged>
NOTICE:  AUDIT: SESSION,6,1,DDL,,bar,[local],pg_regress,,,,,DROP TABLE,TABLE CONSTRAINT,role_test_pkey on public.role_test,DROP TABLE role_test;,<not logged>
NOTICE:  AUDIT: SESSION,6,1,DDL,,bar,[local],pg_regress,,,,,DROP TABLE,INDEX,public.role_test_pkey,DROP TABLE role_test;,<not logged>
